src/main/java/net/minecraftforge/common/util/JsonMerger.java

```diff
@@ -242,92 +252,93 @@ private static void processJsonArrayElement(List<JsonElement> parent, JsonElemen
 
                 if (settings.has("index"))
                 {
+                    if (mode == MergeMode.APPEND)
+                        throw new JsonSyntaxException("Index is noy available in append mode.");
```

Comment:
Not important, but there's a typo in the exception message.
```suggestion
                        throw new JsonSyntaxException("Index is not available in append mode.");
```